src/pages/shared/navigation/navigation.jsx

```diff
@@ -1,56 +1,101 @@
+import "./navigation.css";
+
+// Хуки
 import useClickOutside from "../../../hooks/useClickOutside.js";
 import useWindowWidth from "../../../hooks/useWindowWidth.js";
-import {React, useState, useRef, useEffect } from "react";
+
+// React и хуки
+import { React, useState, useRef, useEffect, useContext } from "react";
+
+// Компоненты
 import Button from "../NavigationButton/Button.jsx";
 import DropDownMenu from "./DropDownMenu/drop_menu.jsx";
-import { Link } from "react-router-dom";
-import UserIco from  "../icons/User_ico.svg"
-import CartIco from  "../icons/Cart_ico.svg"
-import SearchIco from  "../icons/Search_ico.svg"
-import ListIco from  "../icons/TEMP_List_ico.svg"
-import Clear from "../icons/Clear_ico.svg"
 import RegistrationForm from "./RegisterForm/RegisterFormContent.jsx"
 import Modal from "../Modals/Modal.jsx";
-import CartModalContent from "./CartModalContent/CartModalContent.jsx";
+import CartModalContent from './CartModalContent/CartModalContent.jsx';
+
+// Иконки
+import UserIco from "../icons/User_ico.svg";
+import CartIco from "../icons/Cart_ico.svg";
+import SearchIco from "../icons/Search_ico.svg";
+import ListIco from "../icons/TEMP_List_ico.svg";
+import Clear from "../icons/Clear_ico.svg";
+
+// API и контексты
+import { Requests } from "../../../api/axios_queries/requests.js";
+import axiosClient from "../../../api/axios_queries/axios.js";
+import { AppContext } from "../../../context/AppContext.jsx";
+
+// Роутинг
+import { Link, useNavigate } from "react-router-dom";
+
 
 function Navigation(){
     const [isDropOpen, setIsDropOpen] = useState(false);
     const [isClearButtonOpen, setIsClearButtonOpen] = useState(false);
     const [inputText, setInputText] = useState("");
+  
     const [isRegModalOpen, setIsRegModalOpen] = useState(false);
     const [isCartModalOpen, setIsCartModalOpen] = useState(false);
 
+    const { setUserInfo, isUserAuthorized, setIsUserAuthorized} = useContext(AppContext);
+    const navigate = useNavigate();
+
+
     const registerModalRef = useRef(null);
     const cartModalRef = useRef(null);
     const menuRef = useRef(null);
     const menuButtonRef = useRef(null);
 
     useClickOutside([menuRef, menuButtonRef], isDropOpen, () => setIsDropOpen(false));
     const windowWidth = useWindowWidth();
-    
+
+    useEffect(() => {
+        const requests = new Requests(axiosClient);
+        const getUserInfo = async () => {
+            let res
+            try {
+                res = await requests.getUserInfo();
+                setUserInfo(res.data);
+                setIsUserAuthorized(true);
+
+            } catch (error) {
+                if (error.response?.status === 401) {
+                    setIsUserAuthorized(false);
+                    setUserInfo(null);
+                }
+            }
+        }
+            getUserInfo();
+    }, [setIsUserAuthorized, setUserInfo ]);
+
+    // СЛУЖЕБНЫЙ (нужен для вывода юзер даты сразу после того, как она изменилась, потом убрать)
```

Comment:
Language - 1
U didn't put an additional info about the time when do we need to delete it
Or I don't see the point of this part there